tests/global-setup.ts

```diff
@@ -0,0 +1,15 @@
+import {chromium, expect, FullConfig} from '@playwright/test';
+
+async function globalSetup(config: FullConfig) {
+    const browser = await chromium.launch();
+    const page = await browser.newPage();
+    await page.goto(`${config.projects[0].use.baseURL}/admin/login.php`);
+    await page.getByPlaceholder('Password').fill('admin');
+    await page.getByText('Log in').click();
+    await expect(page.getByText('Total queries over last 24 hours')).toBeVisible();
+    // Save signed-in state to 'storageState.json'.
+    await page.context().storageState({ path: 'storageState.json' });
+    await browser.close();
+}
```

Comment:
All this setup does is log into the Pihole a single time and then extracts the cookies so that the different browsers and tests do not also have to login multiple times